pkg/payload/precondition/precondition.go

```diff
@@ -71,25 +72,40 @@ func Summarize(errs []error, force bool) (bool, error) {
 		return false, nil
 	}
 	var msgs []string
+	var isWarning = true
 	for _, e := range errs {
 		if pferr, ok := e.(*Error); ok {
 			msgs = append(msgs, fmt.Sprintf("Precondition %q failed because of %q: %v", pferr.Name, pferr.Reason, pferr.Error()))
+			if !pferr.NonBlockingWarning {
+				isWarning = false
+			}
 			continue
 		}
 		msgs = append(msgs, e.Error())
+
 	}
 	msg := ""
 	if len(msgs) == 1 {
 		msg = msgs[0]
 	} else {
 		msg = fmt.Sprintf("Multiple precondition checks failed:\n* %s", strings.Join(msgs, "\n* "))
+
+		// when there are multiple precondition checks failed we will ignore if one of them is marked warning
+		isWarning = false
```

Comment:
I don't understand what this is about.  If there are two warning messages, won't this code make the summarized condition blocking, when we want to keep it non-blocking?

Reply:
Good catch, I forgot to remove it.

Reply:
Resolved in https://github.com/openshift/cluster-version-operator/commit/481bcde393f244f4e4245ebec6a43941c87626de